psga/common/primer_autodetection.nf

```diff
@@ -0,0 +1,81 @@
+/*
+ * Run: primer-autodetection (use first read only for illumina)
+ */
+process primer_autodetection {
+  publishDir "${params.output_path}/primer_autodetection", mode: 'copy', overwrite: true, pattern: '{*_primer.txt,*_primer_data.csv,*_primer_detection.tsv,trimmomatic.out,*.bowtie2*}'
+
+  tag "${task.index} - ${file_1}"
+
+  input:
+    path file_1
+    val kit
+
+  output:
+    path "*_primer.txt", emit: ch_primer
+    path "*_primer_data.csv", emit: ch_primer_data
+    path "*_primer_detection.tsv", emit: ch_primer_coverage
+    path "trimmomatic.out"
+    path "*.bowtie2*"
+
+  shell:
+  '''
+  kit=!{kit}
+  file_1=!{file_1}
+  sample_id=$( echo ${file_1} | cut -d '.' -f1 )
+  trimmed_sample="trimmed_sample.fastq"
+  crop="!{params.primer_minimum_length}"
+
+  # trim sample reads so that they are long as the shortest primer
+  java -jar /opt/trimmomatic/trimmomatic.jar SE -phred33 ${file_1} ${trimmed_sample} CROP:${crop} 2> trimmomatic.out
```

Comment:
What is this actually doing? If it's just trimming the read do we really need java to do that?

Reply:
java was needed due to the way trimmomatic was installed. With the new docker image, trimmomatic is invoked directly.